VM/src/laux.cpp

```diff
@@ -183,6 +183,23 @@ LUALIB_API double luaL_optnumber(lua_State* L, int narg, double def)
     return luaL_opt(L, luaL_checknumber, narg, def);
 }
 
+LUALIB_API bool luaL_checkboolean(lua_State* L, int narg)
+{
+    // This checks specifically for boolean values, ignoring
+    // all other truthy/falsy values. If the desired result
+    // is true if value is present then lua_toboolean should
+    // directly be used instead.
+    const auto isbool = lua_isboolean(L, narg);
```

Comment:
fwiw we don't use `auto` except for cases when the type is already spelled out; this variable is also redundant, `lua_isboolean` can just be checked by the `if` statement